core/include/detray/navigation/navigator.hpp

```diff
@@ -50,6 +41,8 @@ enum class status {
     e_on_portal = 2,       ///< reached portal surface
 };
 
+static constexpr std::size_t default_cache_size{10u};
```

Comment:
This means that the number of candidates for navigation is limited by 10?

Reply:
Yes, I just set that as the default for now. This can be changed in the navigator type

Reply:
Can we make it 50 or 100? The global memory allocation is not a major factor for the tracking chain time

Reply:
This means the navigator can hold 10 candidates at a time, not that it will not be able to only handle 10 in a volume. Once these 10 candidates are dealt with and no portal was hit, it will put new candidates of the same volume in. That is what the ```insert``` method in this PR does.

According to my observations timing this PR, I think, the size of the allocation actually makes an observable difference in the performance on GPU (CPU seems to be less affected).

Reply:
> This means the navigator can hold 10 candidates at a time, not that it will only be able to handle 10 in a volume

I might need a better explanation. If the navigation windows size is 50 for the surface grid, the navigation with the cache size of 10 will lose 40 potential candidates?

> According to my observations timing this PR, I think, the size of the allocation actually makes an observable difference in the performance on GPU 

Doesn't it only affect the time in detector creation? Or does it also affect the actual run time iterating over events?

Reply:
OK. Sorry for the silly questions

Reply:
No, don't worry, they are not stupid at all! I updated the description to be more detailed